Sources/SwiftIfConfig/StaticBuildConfiguration.swift

```diff
@@ -0,0 +1,402 @@
+    self.targetObjectFileFormats = targetObjectFileFormats
+    self.targetPointerBitWidth = targetPointerBitWidth
+    self.targetAtomicBitWidths = targetAtomicBitWidths
+    self.endianness = endianness
+    self.languageVersion = languageVersion
+    self.compilerVersion = compilerVersion
+  }
+
+  /// The set of custom conditions that are present and can be used with `#if`.
+  ///
+  /// Custom build conditions can be set by the `-D` command line option to
+  /// the Swift compiler. For example, `-DDEBUG` sets the custom condition
+  /// named `DEBUG`, which could be checked with, e.g.,
+  ///
+  /// ```swift
+  /// #if DEBUG
+  /// // ...
+  /// #endif
+  /// ```
+  public var customConditions: Set<String> = []
+
+  /// The set of language features that are enabled.
+  ///
+  /// Features are determined by the Swift compiler, language mode, and other
+  /// options such as `--enable-upcoming-feature`, and can be checked with
+  /// the `hasFeature` syntax, e.g.,
+  ///
+  /// ```swift
+  /// #if hasFeature(VariadicGenerics)
+  /// // ...
+  /// #endif
+  /// ```
+  public var features: Set<String> = []
+
+  /// The set of attributes that are available.
+  ///
+  /// Attributes are determined by the Swift compiler. They can be checked
+  /// with `hasAttribute` syntax, e.g.,
+  ///
+  /// ```swift
+  /// #if hasAttribute(available)
+  /// // ...
+  /// #endif
+  /// ```
+  public var attributes: Set<String> = []
+
+  /// The active target OS names, e.g., "Windows", "iOS".
+  public var targetOSNames: Set<String> = []
+
+  /// The active target architectures, e.g., "x64_64".
+  public var targetArchitectures: Set<String> = []
+
+  /// The active target environments, e.g., "simulator".
+  public var targetEnvironments: Set<String> = []
+
+  /// The active target runtimes, e.g., _ObjC.
+  public var targetRuntimes: Set<String> = []
+
+  /// The active target's pointer authentication schemes, e.g., "arm64e".
+  public var targetPointerAuthenticationSchemes: Set<String> = []
+
+  /// The active target's object file formats, e.g., "COFF"
+  public var targetObjectFileFormats: Set<String> = []
+
+  /// The bit width of a data pointer for the target architecture.
+  ///
+  /// The target's pointer bit width (which also corresponds to the number of
+  /// bits in `Int`/`UInt`) can only be queried with the experimental syntax
+  /// `_pointerBitWidth(_<bitwidth>)`, e.g.,
+  ///
+  /// ```swift
+  /// #if _pointerBitWidth(_32)
+  /// // 32-bit system
+  /// #endif
+  /// ```
+  public var targetPointerBitWidth: Int = 64
+
+  /// The atomic bit widths that are natively supported by the target
+  /// architecture.
+  ///
+  /// This lists all of the bit widths for which the target provides support
+  /// for atomic operations. It can be queried with
+  /// `_hasAtomicBitWidth(_<bitwidth>)`, e.g.
+  ///
+  /// ```swift
+  /// #if _hasAtomicBitWidth(_64)
+  /// // 64-bit atomics are available
+  /// #endif
+  public var targetAtomicBitWidths: [Int] = []
```

Comment:
Why is this an `Array` when eg. `targetEnvironments` is a `Set`?

Reply:
This way, it matches the corresponding requirement in `BuildConfiguration`.